<launch>
  <!-- 공통 인자: 필요시 바꿔서 사용 -->
  <arg name="model" default="traffic_colab.pt"/>
  <arg name="input_topic" default="/image_jpeg/compressed"/>
  <arg name="result_topic" default="/traffic_light/result"/>
  <arg name="result_image_topic" default="/traffic_light/image_raw"/>
  <arg name="conf_thres" default="0.73"/>
  <arg name="iou_thres" default="0.45"/>
  <arg name="max_det" default="300"/>
  <arg name="classes" default="[0,1,2,3,4]"/>
  <arg name="tracker" default="bytetrack.yaml"/>
  <arg name="device" default="cuda"/>

  <!-- 노드 실행 -->
  <node pkg="stop_sequence" type="traffic_light_detection.py" name="traffic_light_detect_node" output="screen">
    <param name="model" value="$(arg model)"/>
    <param name="input_topic" value="$(arg input_topic)"/>
    <param name="result_topic" value="$(arg result_topic)"/>
    <param name="result_image_topic" value="$(arg result_image_topic)"/>

    <param name="conf_thres" value="$(arg conf_thres)"/>
    <param name="iou_thres" value="$(arg iou_thres)"/>
    <param name="max_det" value="$(arg max_det)"/>
    <param name="classes" value="$(arg classes)" type="yaml"/>
    <param name="tracker" value="$(arg tracker)"/>
    <param name="device" value="$(arg device)"/>

    <param name="result_conf" value="true"/>
    <param name="result_line_width" value="1"/>
    <param name="result_font_size" value="1"/>
    <param name="result_labels" value="true"/>
    <param name="result_boxes" value="true"/>
  </node>
</launch>